core/src/actors/mining_manager/handlers.rs

```diff
@@ -0,0 +1,100 @@
+use actix::{ActorFuture, Context, ContextFutureSpawner, Handler, System, WrapFuture};
+
+use super::MiningManager;
+use crate::actors::chain_manager::messages::GetHighestCheckpointBeacon;
+use crate::actors::chain_manager::ChainManager;
+use crate::actors::epoch_manager::messages::EpochNotification;
+
+use witnet_data_structures::chain::{
+    Hash, Hashable, LeadershipProof, Secp256k1Signature, Signature,
+};
+
+use crate::actors::chain_manager::messages::BuildBlock;
+use crate::actors::reputation_manager::messages::ValidatePoE;
+use crate::actors::reputation_manager::ReputationManager;
+use log::{debug, error};
+
+////////////////////////////////////////////////////////////////////////////////////////
+// ACTOR MESSAGE HANDLERS
+////////////////////////////////////////////////////////////////////////////////////////
+
+/// Payload for the notification for all epochs
+#[derive(Clone, Debug)]
+pub struct EveryEpochPayload;
+
+/// Handler for EpochNotification<EveryEpochPayload>
+impl Handler<EpochNotification<EveryEpochPayload>> for MiningManager {
```

Comment:
This handler's logic is too long. Let's see what can be taken out into static functions.